src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.CachedDocumentSearch.cs

```diff
@@ -53,24 +57,30 @@ private static void ClearCachedData()
         }
 
         public async Task SearchCachedDocumentsAsync(
-            Project project,
+            Solution solution,
+            ImmutableArray<Project> projects,
             ImmutableArray<Document> priorityDocuments,
             string searchPattern,
             IImmutableSet<string> kinds,
             Document? activeDocument,
-            Func<INavigateToSearchResult, Task> onResultFound,
+            Func<Project, INavigateToSearchResult, Task> onResultFound,
+            Func<Task> onProjectCompleted,
             CancellationToken cancellationToken)
         {
-            var solution = project.Solution;
+            Contract.ThrowIfTrue(projects.IsEmpty);
+            Contract.ThrowIfTrue(projects.Select(p => p.Language).Distinct().Count() != 1);
+
+            Debug.Assert(priorityDocuments.All(d => projects.Contains(d.Project)));
+
             var onItemFound = GetOnItemFoundCallback(solution, activeDocument, onResultFound, cancellationToken);
 
-            var documentKeys = project.Documents.SelectAsArray(DocumentKey.ToDocumentKey);
+            var documentKeys = projects.SelectMany(p => p.Documents.Select(DocumentKey.ToDocumentKey)).ToImmutableArray();
```

Comment:
>SelectMany [](http://example.com/codeflow?start=40&length=10)

SelectManyAsArray?

Reply:
that makes sense.  can i do in followup if there isn't more significant feedback? :)